sdk/src/azure/core/CMakeLists.txt

```diff
@@ -94,10 +94,18 @@ target_include_directories(az_core
     $<BUILD_INTERFACE:${az_SOURCE_DIR}/sdk/tests/core/inc>
 )
 
-target_link_libraries(az_core
-  PUBLIC
-    ${PAL}
-)
+if (APPLE) # temporary
+  target_link_libraries(az_core
+    PUBLIC
+      ${PAL}
+  )
+else()
+  target_link_libraries(az_core
+    PUBLIC
+      ${PAL}
+      ${MQTT_TRANSPORT}
```

Comment:
Nit: please ensure this isn't built on Windows either for now since we don't have an Schannel-based transport available yet (Mosquitto/Paho are both using OpenSSL AFAIK).